src/components/story-element.js

```diff
@@ -94,12 +95,63 @@ const DEFAULT_TEMPLATES = {
 };
 
 class StoryElementBase extends React.Component {
+
+  constructor(props){
+    super(props);
+    this.observer = null;
+    this.storyElementRef = null;
+  }
+
+  componentDidMount() {
+    this.initiateObserver();
+  }
+
+  componentWillUnmount() {
+    this.destroyObserver();
+  }
+
+  initiateObserver = () => {
+    if(this.props.disableAnalytics === true) return false;
+
+    const options = {
+      root: null,
+      rootMargin: '0px',
+      threshold: 1.0
+    };
+    this.observer = new IntersectionObserver(this.observerCallback, options);
+    this.observer.observe(this.storyElementRef);
+  };
+
+  destroyObserver = () => {
+    this.observer && this.observer.disconnect();
+  };
+
+  observerCallback = entries => {
+    entries.forEach(entry => {
+      if (entry.isIntersecting) {
+        this.emitElementQlitics();
+        this.destroyObserver(); //destroy observer once the call is made
```

Comment:
i think we need to fire it everytime it comes in view.

Reply:
done